tests/pos/choices.scala

```diff
@@ -0,0 +1,103 @@
+import collection.mutable.ListBuffer
+import compiletime.uninitialized
+import scala.util.boundary, boundary.{Label, break}
+
+/** Contains a delimited contination, which can be invoked with `r`esume` */
+class Suspension[+R]:
+  def resume(): R = ???
+object Suspension:
+  def apply[R](): Suspension[R] =
+    ??? // magic, can be called only from `suspend`
```

Comment:
Yes, this looks very close to what we need if there are no native suspensions. The question is, can you do this without having to split higher-order functions on the user side? I.e. can you work with a `map` signature like this:
```scala
extension [A](xs: List[A])
  def map[B](f: A => B): List[B]
```
and have it work for asynchronous functions `f` as well? It seems to me that you will have to duplicate `map`s code to specialize for sync vs async. If you can do this automatically in the code generation, that would be ideal.

Also agree that we need to integrate with cancellation. The strawmans here don't do that yet, since they are just the first baby steps.

Reply:
> which platform it should run on (what is the calling convention on the platform, are tail calls supported, etc.)

I think we are looking for solutions on all platforms: JVM with or without Loom, Scala Native, and Scala JS. This means we might need more than one solution. 

> what are the desired performance characteristics (direct style code, continuation capture, resumption, ec.)

I believe a solution should not slow down direct-style code. Typical use case for resumptions is async code waiting on external events. 

> what style of resumption is desired (resume once, multiple times, only tail, ...)

The use case that must be supported is async, which I believe can be expressed with tail continuations? The rest is good to have but not required if doing so would break some of the other requirements.